Core/src/org/sleuthkit/autopsy/datasourceprocessors/xry/AbstractSingleKeyValueParser.java

```diff
@@ -0,0 +1,165 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.datasourceprocessors.xry;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * Template parse method for reports that make blackboard attributes from a
+ * single key value pair.
+ *
+ * This parse implementation will create 1 artifact per XRY entity.
+ */
+abstract class AbstractSingleKeyValueParser implements XRYFileParser {
+    
+    private static final Logger logger = Logger.getLogger(AbstractSingleKeyValueParser.class.getName());
+
+    private static final char KEY_VALUE_DELIMITER = ':';
+
+    @Override
+    public void parse(XRYFileReader reader, Content parent) throws IOException, TskCoreException {
+        Path reportPath = reader.getReportPath();
+        logger.log(Level.INFO, String.format("INFO: Processing report at [ %s ]", reportPath.toString()));
+
+        while (reader.hasNextEntity()) {
+            String xryEntity = reader.nextEntity();
+            String[] xryLines = xryEntity.split("\n");
+
+            List<BlackboardAttribute> attributes = new ArrayList<>();
+
+            //First line of the entity is the title.
+            if (xryLines.length > 0) {
+                logger.log(Level.INFO, String.format("INFO: Processing [ %s ]", xryLines[0]));
+            }
+
+            String namespace = "";
+            //Process each line, searching for a key value pair or a namespace.
+            //If neither are found, an error message is logged.
+            for (int i = 1; i < xryLines.length; i++) {
+                String xryLine = xryLines[i];
+
+                //Check if the line is a namespace, which gives context to the keys
+                //that follow.
+                if (isNamespace(xryLine)) {
+                    logger.log(Level.INFO, String.format("INFO: Detected XRY "
+                            + "namespace keyword [ %s ]. Applying to all key value pairs following it.", xryLine));
+                    namespace = xryLine.trim();
+                    continue;
+                }
+
+                //Find the XRY key on this line. Assume key is the value between
+                //the start of the line and the first delimiter.
+                int keyDelimiter = xryLine.indexOf(KEY_VALUE_DELIMITER);
+                if (keyDelimiter == -1) {
+                    logger.log(Level.SEVERE, String.format("Expected a key value "
```

Comment:
I am not convinced that these errors rise to the level of SEVERE. A WARNING is the usual level for corrupt data. Are you thinking about this differently?

Also, I would add something distinct to all of the log messages (see prior comment) output by these parsers to make it easier to grep for the messages during field trials.

Reply:
I know. I was thinking of marking them as severe for the time being to make sure these were not missed. I can replace them with warning and prepend "XRY DSP" if you prefer.

Reply:
That sounds good. We definitely want to make it easy to get feedback from the field trials..